### ISSUE TITLE
Earthly seems to misinterpret Windows native path

### ISSUE THREAD

**Issue by GMaxera** [earthly/earthly#1663]
```
earthly.exe version v0.6.7 30a39d9b10c429e5a34cdd753652baf9966a3b63 windows/amd64; Windows 10
```

Trying to use Earthly on Powershell 7.1.3 on Windows 10 and no matter how I specify the path of the build target, Earthly always exit with an error:

**Relative Path and Windows 10 slashes**
```
(osirium-main-official-py38) C:\..\repos\osirium-main> earthly .\testing\unit_and_integration+image

Error: build target: build main: bkClient.Build: failed to solve: resolve build context for target .testingunit_and_integration+image: No Earthfile nor build.earth file found for target .testingunit_and_integration+image
```

**Relative Path and Unix slashes**
```
(osirium-main-official-py38) C:\..\repos\osirium-main> earthly ./testing/unit_and_integration+image

Error: build target: build main: bkClient.Build: failed to solve: resolve build context for target ./testing/unit_and_integration+image: get rel dir for ./testing/unit_and_integration when base git path is C:/Users/gianluca.massera/Developer/repos/osirium-main: stat for [C: Users gianluca.massera Developer repos osirium-main testing unit_and_integration]: CreateFile \C:.: The filename, directory name, or volume label syntax is incorrect.
```

**Absolute Path and Windows 10 slashes**
```
(osirium-main-official-py38) C:\..\repos\osirium-main> earthly C:\Users\gianluca.massera\Developer\repos\osirium-main\testing\unit_and_integration+image

Error: build target: build main: bkClient.Build: failed to solve: resolve build context for target C:Usersgianluca.masseraDeveloperrepososirium-maintestingunit_and_integration+image: failed to get url for cloning: no git match found
```

**Absolute Path and Unix slashes**
```
(osirium-main-official-py38) C:\..\repos\osirium-main> earthly C:/Users/gianluca.massera/Developer/repos/osirium-main/testing/unit_and_integration+image

Error: build target: build main: bkClient.Build: failed to solve: resolve build context for target C:/Users/gianluca.massera/Developer/repos/osirium-main/testing/unit_and_integration+image: failed to get url for cloning: no git match found
```

I have exhausted all options that came up in my mind.
So, I'm wondering what else I can do?

**Comment by brandonSc** [comment]
Thanks for reporting this @GMaxera - I am also seeing the same issue on Windows Powershell. Seems that we should have better support for handling Window-style paths with `\` here.

**Comment by brandonSc** [comment]
A potential area in the code where the fix could be applied: https://github.com/earthly/earthly/blob/main/domain/reference.go#L183-L196

**Comment by brandonSc** [comment]
One workaround for the meantime is to use forward-slash instead of back-slash in the path, for example
```
earthly ./testing/unit_and_integration+image
```
instead of 
```
earthly .\testing\unit_and_integration+image
```

That seems to work for me. 

I see you tried that in your example but got the error ` get rel dir for ./testing/unit_and_integration when base git path is C:/Users/gianluca.massera/Developer/repos/osirium-main`. That error might be a separate issue related to https://github.com/earthly/earthly/issues/1664.

**Comment by GMaxera** [comment]
@brandonSc I've checkout the code and look into Earthly to see if I could help.
I think I may have found the culprit of Windows 10 error, looking in the code I've found this:
```
	// `pathParts` here has lose the root filepath.Separator since built by strings.Split beforehand.
	// so putting heading separator is required to make it absolute again.
	b, err := os.Stat(string(filepath.Separator) + filepath.Join(pathParts[:len(basePathParts)]...))
	if err != nil {
		return "", false, errors.Wrapf(err, "stat for %v", pathParts)
	}
```
The comment says that "putting heading separator is required to make it absolute again" ... but this is wrong on Windows 10 where the native absolute paths starts with a volume label: `C:`

**Comment by brandonSc** [comment]
Good catch @GMaxera and thanks for looking into it 🙌 

There's also this piece of code that wants to treat `\` as an escape key instead of a directory path. For example, it turns `.\testing\unit_and_integration` into `.testingunit_and_integration`. There might also be more places that aren't prepared to deal with `C:` in an absolute path. We'll get this fixed asap.

**Comment by GMaxera** [comment]
I've started an attempt on fixing this and other issues uncovered while I was going further into the build in https://github.com/earthly/earthly/pull/1681

**Comment by brandonSc** [comment]
This is now fixed in [v0.6.9](https://github.com/earthly/earthly/releases/tag/v0.6.9)
